web_ui/CONTRIBUTING.md

```diff
@@ -0,0 +1,132 @@
+# Contributing to Intel Geti Web UI
+
+**Prerequisites**: we recommend developing Geti's UI locally using node v22.12.0 and npm v10.9.0. You will also need to have a Geti server available. Please see our [documentation]() for instructions to install Geti on your hardware.
+
+## Local Setup with npm 
+
+To get started with contributing to the UI first install all node dependencies locally
+
+```sh
+web_ui/ $ npm ci
+```
+
+Next make a `.env` file that includes a `REACT_APP_API_PROXY` variable, you copy over the `.env.example` file,
+```sh
+web_ui/ $ cp .env.example .env
+```
+
+By default `REACT_APP_API_PROXY` is set to `localhost:80`, assuming you have a server running on port 80. We will redirect all requests to `/api` to `localhost:80/api`.
+Assuming you have your Geti server running on `localhost:80` you are now all set to start changing the Geti UI.
+Start up the [Rsbuild](https://rsbuild.dev/) dev server with,
+
+```sh
+web_ui/ $ npm run start
+```
+
+Continue reading to learn more about our linting and testing setup.
+
+## Testing and Quality Control
+
+We heavily rely on Typescript, ESLint, Jest and Playwright to make sure we don't introduce regressions as well as to help us improve the design of our code.
+
+### Typescript
+
+```sh
+npm run test:tsc
+```
+
+Checks if there are no typescript violations. We have a relatively strict typescript configuration and also use `typescript-eslint` for some additional safeguards.
+
+### Check lint rules
+
+```sh
+npm run test:eslint
+npm run test:eslint:fix
+```
+
+Our CI uses `npm run test:eslint` to verify that no eslint rules are violated. For development we often use `npm run test:eslint:fix` to automatically fix these where possible.
+
+#### Cyclic dependencies
+
+In the past we've had issues where cyclic dependencies introduced build regressions, this is why we added an additional step in our CI to check for this.
+Use,
+```sh
+npm run cyclic-deps-check
+```
+
+### Formatting with prettier
+
+We use prettier to automatically format our code,
+
+```
+npm run format:check
+npm run format
+```
+
+Note that `npm run format` will both check and fix any formatting issues.
+
+### Unit testing
+
+```sh
+npm run test:unit
+npm run test:unit:watch
+```
+
+Use the watch command to interactively choose the unit tests you want to run while making code changes.
+Depending on your machine's memory and cpu you may want to add a `--maxWorkers` to prevent jest from using up all your system's resources,
+
+```sh
+npm run test:unit:watch -- --maxWorkers=4
```

Comment:
not sure about this one. I had random failures with it. I think the default (flagless) would be more appropriate

Reply:
We can also use a percentage to be fully safe. Since workers = #cores, we can just tell the user to use maxWorkers=50% just to be safe, but that it can be removed for 100% usage (at their own discretion :D )

Reply:
For me the main issue I sometimes face when running it without `maxWorkers` that it will sometimes hog all of my RAM, especially if I already have half a lot of my RAM being consumed by other processes.
Setting a percentage could help but it could still consume more RAM than I'd like.

The real solution for this would be to improve the memory consumption of our unit tests, but I guess that will require more effort.